File: fp_py/src/main/fruitpunch/native/common/iterator.py
import libfp_core
from fruitpunch.native.native_class import NativeClass

class Iterator(NativeClass):

    def __init__(self):
        NativeClass.__init__(self)
        
        self.store(libfp_core.Iterator())
    
    def go_to(self, i):
        return self.run_native("go_to", i)
    
    def get(self):
        return self.run_native("get")
    
    def next(self):
        val = self.run_native("next")
        if not val:
            raise StopIteration
        return self.get()
    
    def remove(self):
        return self.run_native("remove")
    
NativeClass.class_map[libfp_core.Iterator] = Iterator

File: fp_core/src/main/src/Common/Iterator.cpp
#include <fruitpunch/Common/Iterator.h>


namespace fp_core {
namespace common{

Iterator::Iterator() : Object() {
}


void Iterator::goTo(int i) {
}

boost::shared_ptr<Object> Iterator::get() {
	return boost::shared_ptr<Object>();
}

bool Iterator::next() {
	return false;
}

void Iterator::remove() {
}

}
}

